meteor-core/src/main/java/dev/pixelib/meteor/core/proxy/ProxyInvocHandler.java

```diff
@@ -23,15 +23,18 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
             args = new Object[0];
         }
 
-        // build invocation descriptor
-        Class<?>[] argTypes = new Class<?>[args.length];
-        for (int i = 0; i < args.length; i++)
-            argTypes[i] = args[i].getClass();
+        Class<?>[] argTypes = this.buildInvocationDescriptor(args);
 
         InvocationDescriptor invocationDescriptor = new InvocationDescriptor(namespace, method.getDeclaringClass(), method.getName(), args, argTypes, method.getReturnType());
 
         // wait for response or timeout
         return localInvocationTracker.invokeRemoteMethod(invocationDescriptor);
     }
 
+    private Class<?>[] buildInvocationDescriptor(Object[] args) {
```

Comment:
This code is never used elsewhere and merely maps an array, this would indeed be cleaner except that the method doesn't describe what it does. A more fitting method would be something like `mapArgTypes`, making this contribution feel automated without any real cause.